lib/ui/platform_dispatcher.dart

```diff
@@ -2605,3 +2693,83 @@ class SemanticsActionEvent {
     );
   }
 }
+
+/// Signature for [PlatformDispatcher.onViewFocusChange].
+typedef ViewFocusChangeCallback = void Function(ViewFocusEvent viewFocusEvent);
+
+/// An event for the engine to communicate view focus changes back to the app.
+///
+/// This value will be tipically passed to the [PlatformDispatcher.onViewFocusChange]
+/// callback.
+final class ViewFocusEvent {
+  /// Creates a [ViewFocusChange].
+  const ViewFocusEvent({
+    required this.viewId,
+    required this.focusState,
+    required this.focusDirection,
+  });
+
+  /// The id of the [FlutterView] that experienced a focus change.
+  final int viewId;
+
+  /// The current [ViewFocusState] of the [FlutterView] with the given [viewId].
+  final ViewFocusState focusState;
+
+  /// The current [ViewFocusDirection] of the [FlutterView] with the given [viewId].
+  final ViewFocusDirection focusDirection;
+
+  @override
+  String toString() {
+    return 'ViewFocusEvent(viewId: $viewId, focusState: $focusState, focusDirection: $focusDirection)';
```

Comment:
Consider putting this behind an `assert` so we don't include debug code in production builds.